backend/main.py

```diff
@@ -1,34 +1,56 @@
-from fastapi import FastAPI
+from fastapi import FastAPI, HTTPException,Depends
 from db import get_database
 import json
 from model import *
+from typing import List
 import re
+from fastapi.middleware.cors import CORSMiddleware
 
 app = FastAPI()
 db = get_database()
 collection = db["apps"]
 
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=[""],
+    allow_credentials=True,
+    allow_methods=[""],
+    allow_headers=["*"],
+)
 
-@app.post("/add_item", response_model=str)
-async def add_item(item: AddItem):
-    # Insert the new item into the database
-    await collection.insert_one(item.model_dump())
+
+@app.post("/addItem", response_model=str)
+async def addItem(item: AddItem):
+    # Check if an item with the same name or ID already exists
+    existing_item = await collection.find_one({"$or": [{"name": item.name}, {"id": item.id}]})
+    
+    if existing_item:
+        raise HTTPException(status_code=400, detail="Item with the same name or ID already exists")
+
+    await collection.insert_one({"name": item.name, "icon": item.icon, "id": item.id})
     return "Item added successfully!"
 
 
-@app.get("/get_all_items")
-async def get_all_items():
-    items = await collection.find({}, {"_id": 0}).to_list()
+@app.get("/getAllItems")
+async def getAllItems(name: Optional[str] = None,id: Optional[str] = None):
+    query = {}
+
+    if name:
```

Comment:
You are not handling the edge case that the user might send both query parameter.

Reply:
this one resolved